app-frontend/src/app/pages/lab/run/run.html

```diff
@@ -1,78 +1,59 @@
-                          on-preview="$ctrl.showPreview(data)"
-                          on-share="$ctrl.shareNode(data)"
-                          tool-definition="$ctrl.tool.definition"
-                          tool-parameters="$ctrl.toolParameters">
-    </rf-diagram-container>
-    <div style="position: absolute; top:0; bottom: 0; left: 0; right: 0;" ng-show="$ctrl.isShowingPreview">
-      <rf-map-container map-id="lab-run-preview"></rf-map-container>
-      <div class="preview-control-bar">
-          <button ng-if="$ctrl.previewLayers.length > 1" class="btn btn-default" ng-click="$ctrl.closePreview()">Exit Node Comparison</button>
-          <button ng-if="!$ctrl.previewLayers || $ctrl.previewLayers.length === 1" class="btn btn-default" ng-click="$ctrl.closePreview()">Exit Node Preview</button>
+    <div class="resize-ghost"
+         ng-style="$ctrl.labResizingStyle"
+         ng-show="$ctrl.labResizing"
+         draggable="false">
+      <div class="resize-ghost-text" ng-show="$ctrl.splitPercentX && $ctrl.splitPercentX > 90">
+        Maximize Map
       </div>
-      <div class="preview-info-bar" ng-show="$ctrl.previewLayers.length > 1">
-        <div class="split-container">
-          <button class="btn">{{$ctrl.previewData[0].label}}</button>
-        </div>
-        <div class="split-container">
-          <button class="btn">{{$ctrl.previewData[1].label}}</button>
-        </div>
+      <div class="resize-ghost-text" ng-show="$ctrl.splitPercentX && $ctrl.splitPercentX < 10">
+        Maximize Diagram
       </div>
-      <div class="preview-info-bar" ng-show="$ctrl.previewLayers.length == 1">
-        <div class="split-container">
-          <button class="btn">{{$ctrl.previewData.label}}</button>
-        </div>
+    </div>
+    <div class="lab-right"
+         ng-style="$ctrl.labRightStyle"
+         ng-show="!!$ctrl.showDiagram || !$ctrl.isShowingPreview"
+         draggable="false">
+      <div class="lab-action-controls">
+        <button class="btn btn-light">Add Note</button>
+        <button class="btn btn-secondary" ng-click="$ctrl.createToolRun()">Apply Changes</button>
       </div>
+      <rf-diagram-container ng-if="$ctrl.tool"
+                             class="with-position"
+                             toolrun-valid="!!$ctrl.lastToolRun"
+                             on-preview="$ctrl.showPreview(data)"
+                             on-share="$ctrl.shareNode(data)"
+                             on-parameter-change="$ctrl.onExecutionParametersChange(sourceId, project, band, override)"
+                             tool-definition="$ctrl.tool.definition"
+                             tool-parameters="$ctrl.toolParameters">
+      </rf-diagram-container>
+    </div>
+    <div class="lab-footer"
+         ng-show="$ctrl.warning">
+      <span class="icon-warning"></span>
+      {{$ctrl.warning}}
+      <button class="btn-transparent btn-text" ng-click="$ctrl.clearWarning()">
```

Comment:
This isn't looking quite right
![image](https://user-images.githubusercontent.com/2442245/28374979-4c845b6e-6c74-11e7-8cb9-5faa94b45db3.png)